File: BoyleCohen2008/MatlabSupport/Main_Version/iv.m

%importing data
data = importdata('data/input.csv'); %Not sure if these are the best paramaters...

% Model paramaters

Cmem = 30e-12;
gKS = data(2)*Cmem;
gKF = data(3)*Cmem;
gCa = data(4)*Cmem;
gL = data(5)*Cmem;
VKS = data(6);
VKF = data(7);
VCa = data(8);
VL = 10e-3;
Vhalf_n = data(10);
Vhalf_p = data(11);
Vhalf_q = data(12);
Vhalf_e = data(13);
Vhalf_f = data(14);
Cahalf_h = data(15)*1e-9;
k_n = data(16);
k_p = data(17);
k_q = data(18);
k_e = data(19);
k_f = data(20);
k_h = data(21)*1e-9;
T_n = data(22);
T_p = data(23);
T_q = data(24);
T_e = data(25);
T_f = data(26);
T_Ca = data(27);
alphaCa = data(28);

x = (-60:0.1:80).*1e-3;
numpoints = length(x);
y = zeros(3,numpoints);

for i = 1:numpoints
    IKF = gKF*x_inf(x(i),Vhalf_p,k_p)^4*x_inf(x(i),Vhalf_q,k_q)*(x(i) - VKF);
    IKS = gKS*x_inf(x(i),Vhalf_n,k_n)*(x(i) - VKS);
    ICa = gCa*x_inf(x(i),Vhalf_e,k_e)^2*(x(i) - VCa);
   
    
    y(1,i) = IKF + IKS;
    y(2,i) = ICa;
    y(3,i) = y(1,i) + y(2,i);
end

subplot(122),plot(x*1e3,y(1,:)/Cmem,'k')
hold on
grid on
xlabel('V_{ mem} (mV)')
ylabel('IK (A/F)')
xlim([-70 50])
ylim([-5 40])
subplot(121),plot(x*1e3,y(2,:)/Cmem,'k')
hold on
grid on
xlabel('V_{ mem} (mV)')
ylabel('ICa (A/F)')
xlim([-40 80])
ylim([-8 6])
%plot(x*1e3,y(2,:)/Cmem,'k')

File: BoyleCohen2008/PythonSupport/Main_Version/iv.py
#!/usr/bin/python

# Plotting I-V curves

import sys
import numpy as np
import matplotlib.pyplot as plt

sys.path += '.'
from x_inf import *
from input_vars import *

x = np.arange(-0.060, 0.080, 0.001)
numpoints = len(x)
y = np.zeros((2,numpoints))

for i in range(0,numpoints):
    IKS = gKS * x_inf(x[i], Vhalf_n, k_n) * (x[i] - VKS)
    IKF = gKF * x_inf(x[i], Vhalf_p, k_p)**4 * x_inf(x[i], Vhalf_q, k_q) * (x[i]- VKF)
    ICa = gCa * x_inf(x[i], Vhalf_e, k_e)**2 * (x[i] - VCa)

    y[0][i] = (IKF + IKS) / Cmem
    y[1][i] = ICa  / Cmem


plt.subplot(1,2,1)
plt.plot([round(i * 1e3, 2) for i in x], y[1], 'k', label='$I_{Ca}$')
plt.xlim(-40,80)
plt.ylim(-8,6)
plt.ylabel('I_Ca (A/F)')
plt.xlabel('V (mV)')
plt.grid('on')

plt.subplot(1,2,2)
plt.plot([round(i * 1e3, 2) for i in x], y[0], 'k', label='$I_{K}$')
plt.xlim(-70,50)
plt.ylim(-5,40)
plt.ylabel('I_K (A/F)')
plt.xlabel('V (mV)')
plt.grid('on')

plt.show()
